### ISSUE TITLE
Drop accumulate logits

### ISSUE THREAD

**Issue by kirilklein** [FGA-DIKU/EHR#141]
Accumulating logits in the [_evaluate function ](https://github.com/FGA-DIKU/EHR/blob/1631aee27cfe6e3d7754c66e0e06ae52fd5e4572/corebehrt/modules/trainer/trainer.py#L347) of the trainer is not needed. To avoid getting errors if there is no positive sample in the batch, skip the batch it or handle otherwise gracefully.
